### PR TITLE
Resolve Clip Load - Slate support

### PR DESCRIPTION
**Pull request by jrsndl**
## Changelog Description
Loaded clip should ignore the slate, and be trimmed the same regardless of slate presence.

closes: https://github.com/ynput/OpenPype/issues/6124

#AY-1684

### CODE REVIEW

File: openpype/hosts/resolve/api/plugin.py [hunk 1]
```diff
@@ -435,13 +438,12 @@ def load(self, files):
...
                 handle_end = version_data.get("handleEnd", 0)
                 frame_start_handle = frame_start - handle_start
-                frame_end_handle = frame_start + handle_end
+                frame_end_handle = frame_end + handle_end
                 database_frame_duration = int(
                     frame_end_handle - frame_start_handle + 1
                 )
-                if source_duration >= database_frame_duration:
-                    source_in += handle_start
-                    source_out -= handle_end
```
Comment (BigRoy) [inline openpype/hosts/resolve/api/plugin.py:444]: I'm pretty sure this if statement had a particular reason @jakubjezek001 ?

The large comment above it I think explains it. We need the if statement for representations of a version that exclude the handles even though other representations of the version (and its version data) do have the handles. So we check: if the video is shorter than the database tells us - then we assume this particular representation has already excluded handles.

Technically it's an ingestion issue where the representation is lacking data about its contents - but this was the only sane workaround I believe?

File: openpype/hosts/resolve/api/plugin.py [hunk 2]
```diff
@@ -408,7 +408,11 @@ def load(self, files):
         _clip_property = media_pool_item.GetClipProperty
         source_in = int(_clip_property("Start"))
         source_out = int(_clip_property("End"))
-        source_duration = int(_clip_property("Frames"))
+
+        # Trim clip start if slate is present
+        if "slate" in self.data["versionData"]["families"]:
+            source_in += 1
+        source_duration = source_out - source_in + 1
```
Comment (BigRoy) [inline openpype/hosts/resolve/api/plugin.py:415]: Why would this be more reliable then the `Frames` property? Couldn't we do `source_duration -= 1` for the `slate` case? :)

File: openpype/hosts/resolve/api/lib.py [hunk 3]
```diff
@@ -279,6 +279,10 @@ def create_timeline_item(
     # timing variables
     if all([timeline_in, source_start, source_end]):
         fps = timeline.GetSetting("timelineFrameRate")
+        # Strangely, Resolve seem to output '23' instead of 23.976
+        if fps == '23':
+            fps = 23.976
+
```
Comment (jakubjezek001) [inline openpype/hosts/resolve/api/lib.py:285]:
```suggestion
```

I was testing it on latest Resolve  (18.6) and it was returning `23.976`

**Review by jakubjezek001 — APPROVED** [review]
LGTM

File: openpype/hosts/resolve/api/plugin.py [hunk 4]
```diff
@@ -409,6 +409,11 @@ def load(self, files):
         source_in = int(_clip_property("Start"))
         source_out = int(_clip_property("End"))
         source_duration = int(_clip_property("Frames"))
+
```
Comment (jakubjezek001) [inline openpype/hosts/resolve/api/plugin.py:412]:
```suggestion

```

File: openpype/hosts/resolve/api/plugin.py [hunk 5]
```diff
@@ -409,6 +409,10 @@ def load(self, files):
         source_in = int(_clip_property("Start"))
         source_out = int(_clip_property("End"))
         source_duration = int(_clip_property("Frames"))
+        # Trim clip start if slate is present
```
Comment (jakubjezek001) [inline openpype/hosts/resolve/api/plugin.py:412]:
```suggestion
        # Trim clip start if slate is present
```

**Pull request merged**